corehq/apps/app_manager/detail_screen.py

```diff
@@ -311,7 +311,7 @@ def fields(self):
             # since case list optimizations are only available for versions higher than the
             # ones needed for multi sort, it's safe to assume that case list optimizations
             # should only be added only when multi sort is enabled
-            if self.app.supports_case_list_optimizations:
+            if self.app.supports_case_list_optimizations and self.column.supports_optimizations:
```

Comment:
just a way to strengthen the check to add optimizations on the build files only when relevant.

Reply:
Good point.

I am just thinking of an edge case where someone added the optimizations and then dropped the app version?

Reply:
hmm, yeah, unlikely but possible. Given that you explained that a misconfigured app will break on mobile, I support being cautious here.